src/main/java/org/maxgamer/quickshop/util/updater/impl/JenkinsUpdater.java

```diff
@@ -129,24 +130,21 @@ public void install(byte[] bytes) throws IOException {
         if (plugins == null) {
             throw new IOException("Can't get the files in plugins folder");
         }
-        File quickshop = null;
+        File newJar = new File(pluginFolder, "QuickShop" + UUID.randomUUID().toString().replace("-", "") + ".jar");
+
         for (File plugin : plugins) {
-            try {
+            try { //Delete all old jar files
                 PluginDescriptionFile desc = QuickShop.getInstance().getPluginLoader().getPluginDescription(plugin);
                 if (!desc.getName().equals(QuickShop.getInstance().getDescription().getName())) {
                     continue;
                 }
-                Util.debugLog("Selected: " + plugin.getPath());
-                quickshop = plugin;
-                break;
+                Util.debugLog("Deleting: " + plugin.getPath());
+                plugin.delete();
```

Comment:
Delete itself will making new class need to load broken, I suggest just replace jar

Reply:
Use `update` folder may will be a good idea.
Btw we need to resolve the problem that finding jars from the `plugins` folder, it will be a goal in the next version, not this.